xwiki-platform-core/xwiki-platform-web/xwiki-platform-web-war/src/main/webapp/resources/uicomponents/dashboard/dashboard.js

```diff
@@ -375,24 +375,35 @@ XWiki.Dashboard = Class.create( {
   onEditGadgetClick : function(event) {
     var gadget = event.element().up('.gadget');
 
+    /**
+     * Finds the direct child element of the given element with the specified class name.
+     *
+     * @param {Element} element - the element to search for the child in
+     * @param {string} className - the class name of the child element to find
+     * @returns {Element|undefined} the direct child element with the specified class name, or undefined if not found
+     */
+    function findDirectChildWithClass(element, className) {
+      return element.immediateDescendants().filter((e) => e.hasClassName(className)).first();
```

Comment:
Couldn't find a simpler way to write this, but if anybody has a suggestion I take it.

Reply:
I think you could use Jquery, but it's barely used in this script, so it seems to me like your way is correct
It would be something like :

> $(element).children('.'+className).first().get()

Reply:
The best practice is definitely to switch to jquery/vanillajs to remove prototype. But I would say just changing this single selector wouldn't be beneficial.